lib/msal-browser/docs/shr-server-nonce.md

```diff
@@ -0,0 +1,263 @@
+
+### Acquiring initial nonce
+
+1. The first step to acquire a server-generated nonce is to make an authorized request to the resource. In this authorized request, a `PoP Token` will be added to the `Authorization` header, but said `PoP Token` will not include a valid nonce.
+
+
+```javascript
+
+let shrNonce = ""; // Globally scoped variable
+
+// 1. Configure PoP Token Request without a valid SHR Nonce
+const popTokenRequest = {
+    scopes: ["User.Read"],
+    authenticationScheme: msal.AuthenticationScheme.POP,
+    resourceRequestMethod: "POST",
+    resourceRequestUri: "YOUR_RESOURCE_ENDPOINT"
+    shrNonce: shrNonce // SHR Nonce is invalid as an empty string at this point
+};
+
+
+ // Get PoP token to make authenticated request
+const shr = await publicClientApplication.acquireTokenSilent(popTokenRequest);
+
+// Set up PoP Resource request
+const reqHeaders = new Headers();
+const authorizationHeader = `PoP ${shr}`;
+
+headers.append("Authorization", authorizationHeader);
+
+const options = {
+    method: method,
+    headers: headers
+};
+```
+
+2. Once the request has been set up and the `SHR` has been added to the Authorization header, the request is `POST`ed to the resource. At this point, without a valid server nonce in the `PoP Token`/`SHR`, the resource should respond with a `401 Unauthorized` HTTP error, which will have a `WWW-Authenticate` header containing the first valid server nonce as one of it's challenges.
+
+```typescript
+// Make call to resource with SHR
+return fetch(resourceEndpointData.endpoint, options)
+    .then(response => response.json())
+    .then(response => {
+        // At this point, the response will be a 401 Error, so ignore the success case for now
+    })
+    .catch(error => {
+        // This error will be a `401 Unauthorized` error, containing a WWW-Authenticate header with an error message such as "nonce_missing" or "nonce_malformed"
+        // The correct way to handle this scenario is shown in the following step.
+});
+```
+
+3. In order to extract the server nonce from said `WWW-Authenticate` header, MSAL exposes the `AuthenticationHeaderParser` class, which includes the `getShrNonce` API that will parse the server nonce out of the authentication headers it comes in:
+
+```typescript
+import { PublicClientApplication, AuthenticationHeaderParser } from "@azure/msal-browser";
+
+...
+
+// Make call to resource with SHR
+return fetch(resourceEndpointData.endpoint, options)
+    .then(response => response.json())
+    .then(response => {
+        // At this point, the response will be a 401 Error, so ignore the success case for now
+    })
+    .catch(error => {
+        // If error is 401 Unauthorized, contains WWW-Authenticate and error code is nonce_missing or nonce_expired
```

Comment:
Can you add this logic here for completeness?

Reply:
Will do